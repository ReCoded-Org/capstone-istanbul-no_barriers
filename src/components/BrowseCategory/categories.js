import education from "./icons/education.png";
import food from "./icons/food.png";
import health from "./icons/health.png";
import housing from "./icons/housing.png";
import jobs from "./icons/jobs.png";

const categories = [
  { icon: education, title: "homePage.browser.categories.education", id: "1" },
  { icon: food, title: "homePage.browser.categories.food", id: "2" },
  { icon: health, title: "homePage.browser.categories.health", id: "3" },
  { icon: housing, title: "homePage.browser.categories.housing", id: "4" },
  { icon: jobs, title: "homePage.browser.categories.jobs", id: "5" },
];

export default categories;
