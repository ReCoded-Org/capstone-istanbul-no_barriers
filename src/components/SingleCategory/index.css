:root {
  --text-browse-color: #eff4f6;
  --hover-browse-color: #074081;
}

.catogoryIcon {
  height: 5rem;
  width: auto;
}

.iconTitle {
  color: var(--text-browse-color);
  cursor: pointer;
}

.iconTitle:hover {
  color: var(--hover-browse-color);
}
