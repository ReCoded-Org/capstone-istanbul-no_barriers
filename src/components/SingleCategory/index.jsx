import React from "react";
import { Col } from "react-bootstrap";
import "./index.css";
import { useTranslation } from "react-i18next";

const SingleCategory = ({ icon, title }) => {
  const { t } = useTranslation();

  return (
    <Col className="categoryComponent">
      <img src={icon} alt="category icon" className="catogoryIcon" />
      <p className="iconTitle">{t(title)}</p>
    </Col>
  );
};

export default SingleCategory;
