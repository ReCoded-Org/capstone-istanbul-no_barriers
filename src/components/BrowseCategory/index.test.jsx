import React from "react";
import renderer from "react-test-renderer";
import categories from "./categories";
import SingleCategory from "../SingleCategory/index";

it("Browser renders correctly", () => {
  const tree = renderer
    .create(
      categories.map((category) => (
        <SingleCategory
          icon={category.icon}
          title={category.title}
          key={category.id}
        />
      ))
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
