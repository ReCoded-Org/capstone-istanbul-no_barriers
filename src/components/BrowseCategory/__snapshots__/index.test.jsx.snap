// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser renders correctly 1`] = `
Array [
  <div
    className="categoryComponent col"
  >
    <img
      alt="category icon"
      className="catogoryIcon"
      src="education.png"
    />
    <p
      className="iconTitle"
    >
      homePage.browser.categories.education
    </p>
  </div>,
  <div
    className="categoryComponent col"
  >
    <img
      alt="category icon"
      className="catogoryIcon"
      src="food.png"
    />
    <p
      className="iconTitle"
    >
      homePage.browser.categories.food
    </p>
  </div>,
  <div
    className="categoryComponent col"
  >
    <img
      alt="category icon"
      className="catogoryIcon"
      src="health.png"
    />
    <p
      className="iconTitle"
    >
      homePage.browser.categories.health
    </p>
  </div>,
  <div
    className="categoryComponent col"
  >
    <img
      alt="category icon"
      className="catogoryIcon"
      src="housing.png"
    />
    <p
      className="iconTitle"
    >
      homePage.browser.categories.housing
    </p>
  </div>,
  <div
    className="categoryComponent col"
  >
    <img
      alt="category icon"
      className="catogoryIcon"
      src="jobs.png"
    />
    <p
      className="iconTitle"
    >
      homePage.browser.categories.jobs
    </p>
  </div>,
]
`;
